common/plugins/org.polarsys.capella.common.re.ui/src/org/polarsys/capella/common/re/ui/menu/RecDynamicMenu.java

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.widgets.Menu;
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.ui.ISelectionService;
+import org.eclipse.ui.PlatformUI;
+import org.polarsys.capella.common.re.CatalogElement;
+import org.polarsys.capella.common.re.helpers.ReplicableElementExt;
+import org.polarsys.capella.core.model.handler.helpers.CapellaAdapterHelper;
+import org.polarsys.capella.core.platform.sirius.ui.navigator.actions.ShowInDiagramAction;
+
+public class RecDynamicMenu extends ContributionItem {
+  public RecDynamicMenu() {
+  }
+
+  public RecDynamicMenu(String id) {
+    super(id);
+  }
+
+  @Override
+  public void fill(Menu menu, int index) {
+    ISelectionService selectionService = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getSelectionService();
+    ISelection theSelection = selectionService.getSelection();
+
+    if (!(theSelection instanceof IStructuredSelection)) {
+      return;
+    }
+    IStructuredSelection selection = (IStructuredSelection) theSelection;
+    Collection<CatalogElement> commonRecs = getCommonRecs(selection);
+
+    for (CatalogElement commonRec : commonRecs) {
+      EList<EObject> referencedElements = commonRec.getReferencedElements();
+      IStructuredSelection newSelection = new StructuredSelection(referencedElements);
+      if (!referencedElements.isEmpty()) {
+        MenuItem menuItem = new MenuItem(menu, SWT.CHECK, index);
+        menuItem.setText("Elements of " + commonRec.getName());
+        menuItem.addSelectionListener(new SelectionListener() {
+          @Override
+          public void widgetSelected(SelectionEvent e) {
+            ShowInDiagramAction action = new ShowInDiagramAction();
+            action.selectionChanged(newSelection);
+            action.run();
+          }
+
+          @Override
+          public void widgetDefaultSelected(SelectionEvent e) {
+          }
+        });
+      }
+    }
+
+  }
+
+  public static Collection<CatalogElement> getCommonRecs(IStructuredSelection selection) {
+    LinkedHashSet<EObject> selectedElements = new LinkedHashSet<EObject>();
+    for (Object aSelectedElement : selection.toArray()) {
+        if (aSelectedElement instanceof GraphicalEditPart) {
+          EObject semantic = CapellaAdapterHelper.resolveBusinessObject(((GraphicalEditPart) aSelectedElement));
+          selectedElements.add(semantic);
+        } else {
+          return new ArrayList<CatalogElement>();
+        }
+      }
+
+    HashMap<EObject, Collection<CatalogElement>> mapOfRecs = new HashMap<EObject, Collection<CatalogElement>>();
+    for (EObject anElement : selectedElements) {
+      Collection<CatalogElement> replicablesElements = ReplicableElementExt.getReferencingReplicableElements(anElement);
```

Comment:
Here, we need to filter only the "REC" 
https://github.com/eclipse/capella/blob/409a90c4a2ed6c9c355badee0d068c32ec47f195/common/plugins/org.polarsys.capella.common.re/src/org/polarsys/capella/common/re/helpers/ReplicableElementExt.java#L34-L45

vs

https://github.com/eclipse/capella/blob/409a90c4a2ed6c9c355badee0d068c32ec47f195/common/plugins/org.polarsys.capella.common.re.ui/src/org/polarsys/capella/common/re/ui/queries/CatalogElementOrigin.java#L48-L58

better to use new CatalogElementOrigin().compute t